File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/configuration/model/JobDepartment.java
package dev.reuss.tmdb.domain.configuration.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * TMDB jobs grouped by department.
 *
 * @param department department name
 * @param jobs       jobs in the department
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record JobDepartment(
        String department,
        List<String> jobs
) implements TmdbModel {

    public JobDepartment {
        jobs = TmdbCollections.list(jobs);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/configuration/model/JobDepartment.kt
package dev.reuss.tmdb.domain.configuration.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB jobs grouped by department.
 *
 * @property department department name
 * @property jobs jobs in the department
 */
@JvmRecord
data class JobDepartment(
    @all:JsonProperty("department")
    val department: String?,
    @all:JsonProperty("jobs")
    val jobs: List<String> = emptyList(),
) : TmdbModel
